server.js

```diff
@@ -7,22 +7,145 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
+
+const Thought = mongoose.model('Thought', {
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  },
+  name: {
+    type: String,
+    required: false,
+    default: "anonymous",
+    maxlength: 50
+  },
+  tag: {
+    type: String,
+  }
+})
+
+//SEEDING FOR ADDING NEW DATA. When do I need to use/do this??
+// if (process.env.RESET_DB) {
+//   console.log('Resetting database')
+//   const seedDatabase = async () => {
+//     await Thought.deleteMany({})
+
+//     thought.forEach(() => {
+//       new Thought().save()
+//     })
+//   }
+//   seedDatabase()
+// }
+
```

Comment:
You would use this when you are going to move the data in the database to a new server, or when you want to save/restore the thoughts that are already in the database.  There are more ways to backup/restore.

We were using this code to populate the database with already existing data from a JSON file.  However, for this project there was no pre-existing data